WcaOnRails/lib/database_dumper.rb

```diff
@@ -0,0 +1,447 @@
+      public_condition: "",
+      column_sanitizers: {
+        "event_id" => :keep,
+        "id" => :keep,
+        "user_id" => :keep,
+      }.freeze,
+    }.freeze,
+    "users" => {
+      public_condition: "",
+      column_sanitizers: {
+        "avatar" => :keep,
+        "confirmation_sent_at" => "NULL",
+        "confirmation_token" => "NULL",
+        "confirmed_at" => :keep,
+        "country_iso2" => :keep,
+        "created_at" => :keep,
+        "current_sign_in_at" => :keep,
+        "current_sign_in_ip" => "''",
+        "delegate_id_to_handle_wca_id_claim" => :keep,
+        "delegate_status" => :keep,
+        "dob" => "'1954-12-04'",
+        "email" => "CONCAT(id, '@worldcubeassociation.org')",
+        "encrypted_password" => "''",
+        "gender" => :keep,
+        "id" => :keep,
+        "last_sign_in_at" => :keep,
+        "last_sign_in_ip" => "''",
+        "location_description" => :keep,
+        "name" => :keep,
+        "notes" => "''",
+        "pending_avatar" => "NULL",
+        "phone_number" => "NULL",
+        "preferred_locale" => "NULL",
+        "region" => :keep,
+        "remember_created_at" => "NULL",
+        "reset_password_sent_at" => "NULL",
+        "reset_password_token" => "NULL",
+
+        "results_notifications_enabled" => :keep,
+        "saved_avatar_crop_h" => :keep,
+        "saved_avatar_crop_w" => :keep,
+        "saved_avatar_crop_x" => :keep,
+        "saved_avatar_crop_y" => :keep,
+        "saved_pending_avatar_crop_h" => :keep,
+        "saved_pending_avatar_crop_w" => :keep,
+        "saved_pending_avatar_crop_x" => :keep,
+        "saved_pending_avatar_crop_y" => :keep,
+        "senior_delegate_id" => :keep,
+        "sign_in_count" => "1",
+        "unconfirmed_email" => "NULL",
+        "unconfirmed_wca_id" => :keep,
+        "updated_at" => :keep,
+        "wca_id" => :keep,
+      }.freeze,
+    }.freeze,
+    "vote_options" => :discard,
+    "votes" => :discard,
+  }.freeze
+
+  def self.development_dump(dump_filename)
+    dump_db_name = "wca_development_db_dump"
+    ActiveRecord::Base.connection.execute("DROP DATABASE IF EXISTS #{dump_db_name}")
+    ActiveRecord::Base.connection.execute("CREATE DATABASE #{dump_db_name} DEFAULT CHARACTER SET utf8 DEFAULT COLLATE utf8_unicode_ci")
+    self.mysql("SOURCE #{Rails.root.join('db', 'structure.sql')}", dump_db_name)
+    TABLE_SANITIZERS.each do |table_name, table_sanitizer|
+      next if table_sanitizer == :discard
+
+      column_expressions = table_sanitizer[:column_sanitizers].map do |column_name, column_sanitizer|
+        column_sanitizer == :keep ? "#{table_name}.#{column_name}" : "#{column_sanitizer} as #{column_name}"
+      end.join(", ")
+
+      ActiveRecord::Base.connection.execute("DROP TABLE IF EXISTS #{dump_db_name}.#{table_name}")
+      create_table_sql = "CREATE TABLE #{dump_db_name}.#{table_name} SELECT #{column_expressions} FROM #{table_name} #{table_sanitizer[:public_condition]}"
```

Comment:
`#{table_sanitizer[:public_condition] || ''}` could get rid of the 20 of 27 `public_condition: ""`